ompi/attribute/attribute.c

```diff
@@ -317,9 +319,11 @@ do { \
         /* MPI-2 Fortran-style */ \
         else { \
             MPI_Aint in, out;                        \
+            MPI_Fint attr_##type##_f;                \
             in = translate_to_fortran_mpi2(in_attr); \
+            attr_##type##_f = OMPI_INT_2_FINT(((ompi_##type##_t *)old_object)->attr_##type##_f); \
             (*((keyval_obj->copy_attr_fn).attr_mpi2_fortran_copy_fn)) \
-                (&(((ompi_##type##_t *)old_object)->attr_##type##_f), \
+                (&attr_##type##_f, \
```

Comment:
Seems like all these hunks should be part of 2dba1ec26fa95c2d63434ffe5499998b219a4a7c, too.  I think you can either squash 2dba1ec26fa95c2d63434ffe5499998b219a4a7c and this commit together, or separate out all the `attr_##type##_f` changes into its own commit.